components/clp-py-utils/clp_py_utils/s3_utils.py

```diff
@@ -124,13 +190,22 @@ def s3_put(
         )
 
     config = Config(retries=dict(total_max_attempts=total_max_attempts, mode="adaptive"))
+
+    aws_profile = s3_config.get_profile()
     aws_access_key_id, aws_secret_access_key = s3_config.get_credentials()
 
-    my_s3_client = boto3.client(
+    aws_session = None
+    if aws_profile is not None:
+        aws_session = boto3.Session(profile_name=aws_profile)
+    else:
+        aws_session = boto3.Session(
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            region_name=s3_config.region_code,
+        )
```

Comment:
this is similar to line 133 ~ 144 but written in a slightly different way. can we factor this out as a helper function that eventually returns a s3_client?